targon/internal/targon/callbacks.go

```diff
@@ -110,62 +110,52 @@ func AddBlockCallbacks(v *boilerplate.BaseChainSubscriber, c *Core) {
 		}
 		sendDailyGPUSummary(c, h)
 	})
-	// Set Weights
+	// Set Weights and sync to MongoDB
 	v.AddBlockCallback(func(h types.Header) {
 		if h.Number%360 != 0 || len(c.MinerNodes) == 0 {
 			return
 		}
-		if c.Deps.Mongo != nil {
-			minerInfo := MinerInfo{Core: c, Block: int(h.Number)}
-			err := SyncMongo(c, minerInfo)
-			if err != nil {
-				c.Deps.Log.Errorw("failed syncing to mongo", "error", err)
-			}
-		}
-	})
-	// store emissions to MongoDB
-	v.AddBlockCallback(func(h types.Header) {
-		if h.Number%360 != 0 || len(c.MinerNodes) == 0 || c.Deps.Mongo == nil {
-			return
-		}
-
 		uids, scores, err := getWeights(c)
 		if err != nil {
-			c.Deps.Log.Errorw("Failed getting weights for emissions", "error", err)
+			c.Deps.Log.Errorw("Failed getting weights for MongoDB sync", "error", err)
```

Comment:
This is inaccurate, its getting weights for both mongo sync and setting weights. Just say `failed getting weights` with the error